docs/src/rules/no-unassigned-vars.md

```diff
@@ -0,0 +1,107 @@
+---
+title: no-unassigned-vars
+rule_type: problem
+related_rules:
+- prefer-const
+- init-declarations
+---
+
+
+This rule flags `let` or `var` declarations that have no initializer and are never assigned any value, but are read/used in the code. These variables will always be `undefined` and are likely a programmer error.
+
+This helps catch misleading patterns where a `let` appears mutable but is effectively constant and never written to.
```

Comment:
Let's remove this. I think this wording is confusing and doesn't add anything to the rule description.